### PR TITLE
Enable mlflow logger

### PR DESCRIPTION
**Pull request by whrichd**
Signed-off-by: whrichd <trabeitwrq@gmail.com>

# What does this PR do ?

MLFlow is already a dependancy in Pytorch Lightning, so some small changes enables MLFlow tracking, a free alternative to WnB.

**Collection**: utils

# Changelog 
- Changed exp_manager.py

# Usage
* Add the MLflow logger just as how you would add the weighs and biases logger, in the config files.

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
- [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by okuchaiev** [comment]
@whrichd can you please fix code style? 
```
pip install -r requirements/requirements_test.txt
```
and then
```
python setup.py style --fix
```

**Comment by whrichd** [comment]
Hi @okuchaiev I tried to check the style once before posting this, but it found so many issues outside the scope of this PR so I just left it. How to work around this? Or am I missing some steps?

File: nemo/utils/exp_manager.py [hunk 1]
```diff
@@ -111,6 +111,8 @@ class ExpManagerConfig:
     summary_writer_kwargs: Optional[Dict[Any, Any]] = None
     create_wandb_logger: Optional[bool] = False
     wandb_logger_kwargs: Optional[Dict[Any, Any]] = None
+    create_mlflow_logger: Optional[bool] = True
+    mlflow_logger_kwargs: Optional[Dict[Any, Any]] = None
```
Comment (titu1994) [inline nemo/utils/exp_manager.py:115]: Please create a date class similar to Wandb one for default args and validation

Comment (whrichd) [reply nemo/utils/exp_manager.py:126]: Just realized that you meant changing True to False...

Comment (titu1994) [reply nemo/utils/exp_manager.py:126]: I meant having a dataclass with defaults that are disabled but allows users to validate their arguments in the yaml config. See wandb dataclass used here c

Comment (whrichd) [reply nemo/utils/exp_manager.py:126]: Sorry I couldn't find the dataclass that you are talking about...did you mean to link something in the word `here`?

Comment (titu1994) [reply nemo/utils/exp_manager.py:126]: Check how the config here -https://github.com/NVIDIA/NeMo/blob/main/nemo/utils/exp_manager.py#L72

Maps to the default config here - https://github.com/NVIDIA/NeMo/blob/main/nemo/utils/exp_manager.py#L127

Sane with ema - https://github.com/NVIDIA/NeMo/blob/main/nemo/utils/exp_manager.py#L138

It provides a default config that users should use and validate their yaml against

Comment (titu1994) [reply nemo/utils/exp_manager.py:126]: I see wandb is left as Dict[any, any] which is sad, but at least the arguments are well known in our yaml files. MLFlow logger - we don't know what arguments to provide

Comment (whrichd) [reply nemo/utils/exp_manager.py:126]: I see! Thanks, will change when I get time

**Review by titu1994 — APPROVED** [review]
Looks fantastic, thanks for the help !

File: nemo/utils/exp_manager.py (hunk 1 above)
Comment (titu1994) [reply nemo/utils/exp_manager.py]: Sorry for delay, approved ! Looks great !

**Comment by titu1994** [comment]
Sorry for the delay, forgot to merge

**Pull request merged**